src/domains/tasks/ViewOffers/components/test/ViewOffersPage.test.tsx

```diff
@@ -1,10 +1,125 @@
-import { screen } from "@testing-library/react";
-import { ViewOffersPage } from "../ViewOffersPage";
+import { act, screen } from "@testing-library/react";
+import { rest } from "msw";
+import { server } from "src/mocks/server";
+
+import { ViewOffersPage, ViewOffersPageProps } from "../ViewOffersPage";
 import { renderWithPolarisTestProvider } from "src/test/utils";
+import {
+  ApiEventsProvider,
+  InitialApiEventsState,
+} from "src/common/ApiResponse/ApiEventsContext";
+
+import { getOffersUrl } from "@Tasks/ViewOffers/api";
+import userEvent from "@testing-library/user-event";
+import { getTaskURL } from "@Billing/Checkout/hooks";
+
+type setupRenderOptions = {
+  componentProps?: ViewOffersPageProps;
+  initialProviderState?: InitialApiEventsState;
+};
 
-test("smoke test if it renders", () => {
-  renderWithPolarisTestProvider(<ViewOffersPage status="authenticated" />);
+async function setupRender({
+  initialProviderState,
+  componentProps,
+}: setupRenderOptions = {}) {
+  return renderWithPolarisTestProvider(
+    <ApiEventsProvider initialState={initialProviderState}>
+      <ViewOffersPage
+        status={
+          componentProps?.status ? componentProps.status : "unauthenticated"
+        }
+      />
+    </ApiEventsProvider>
+  );
+}
+
+test("smoke test if it renders", async () => {
+  const componentProps = { status: "authenticated" } as const;
+
+  await act(() => {
+    setupRender({ componentProps });
+  });
 
   const viewOffersPage = screen.getByLabelText(/^view offers page$/i);
   expect(viewOffersPage).toBeTruthy();
 });
+
+test("loads a list of OfferItems", async () => {
+  const componentProps = { status: "authenticated" } as const;
+
+  await act(() => {
+    setupRender({ componentProps });
+  });
+
+  const offerItems = screen.getAllByRole("heading", { name: /^offer$/i });
+  expect(offerItems.length).toBeGreaterThan(1);
+});
+
+test("displays error notification when getOffersOfProviderQuery fails", async () => {
+  const testErrorMessage = "THIS IS A TEST FAILURE";
+  server.use(
+    rest.get(getOffersUrl, async (req, res, ctx) => {
+      return res(ctx.status(500), ctx.json({ message: testErrorMessage }));
+    })
+  );
+
+  await act(() => {
+    setupRender();
+  });
```

Comment:
Having had to wrap the `setupRender` inside an `act` made it hard to extract the returned container.

Perhaps I should move `act` inside my `setupRender` function instead. It won't look pretty, but at least can further refactor our test cases. 

i.e. Moving the `user` event object into `setupRender` as well.